src/lib.rs

```diff
@@ -81,45 +76,31 @@ pub(crate) fn calc_d1d2(inputs: &Inputs) -> Result<(f32, f32), String> {
 /// # Requires
 /// s, k, r, q, t, sigma
 /// # Returns
-/// Tuple (f32, f32) of (nd1, nd2)
-pub(crate) fn calc_nd1nd2(inputs: &Inputs) -> Result<(f32, f32), String> {
-    let nd1nd2 = {
-        let d1d2 = calc_d1d2(inputs)?;
-
-        let n: Normal = Normal::new(N_MEAN as f64, N_STD_DEV as f64).unwrap();
-
-        let num_cast_err: String = "Failed to cast f64 to f32".into();
-        // Calculates the nd1 and nd2 values
-        // Checks if OptionType is Call or Put
-        match inputs.option_type {
-            OptionType::Call => (
-                NumCast::from(n.cdf(NumCast::from(d1d2.0).ok_or(&num_cast_err)?))
-                    .ok_or(&num_cast_err)?,
-                NumCast::from(n.cdf(NumCast::from(d1d2.1).ok_or(&num_cast_err)?))
-                    .ok_or(&num_cast_err)?,
-            ),
-            OptionType::Put => (
-                NumCast::from(n.cdf(NumCast::from(-d1d2.0).ok_or(&num_cast_err)?))
-                    .ok_or(&num_cast_err)?,
-                NumCast::from(n.cdf(NumCast::from(-d1d2.1).ok_or(&num_cast_err)?))
-                    .ok_or(&num_cast_err)?,
-            ),
-        }
+/// Tuple (f64, f64) of (nd1, nd2)
+pub(crate) fn calc_nd1nd2(inputs: &Inputs) -> Result<(f64, f64), String> {
+    let (d1, d2) = calc_d1d2(inputs)?;
+
+    let n: Normal = Normal::new(0.0, 1.0).unwrap();
+
+    // Calculates the nd1 and nd2 values
+    // Checks if OptionType is Call or Put
+    let nd1nd2 = match inputs.option_type {
+        OptionType::Call => (n.cdf(d1), n.cdf(d2)),
+        OptionType::Put => (n.cdf(-d1), n.cdf(-d2)),
     };
     Ok(nd1nd2)
 }
 
 /// Calculates the n probability density function (PDF) for the given input.
 /// # Returns
-/// f32 of the value of the n probability density function.
-pub(crate) fn calc_npdf(x: f32) -> f32 {
-    let d: f32 = (x - N_MEAN) / N_STD_DEV;
-    (-HALF * d * d).exp() / (SQRT_2PI * N_STD_DEV)
+/// f64 of the value of the n probability density function.
+pub(crate) fn calc_npdf(x: f64) -> f64 {
+    (-0.5 * x * x).exp() * FRAC_1_SQRT_2PI
```

Comment:
I understand the need to drop unnecessary instructions, but it might conflict with the formula. I think it is better to keep the same formula in both the code and the docs.

In this case, the compiler will do the same as you have done.

Reply:
yeah, this wasn't so much compiler optimization as wanting the function to read cleaner, I'm sure the compiler instructions are essentially the same.